molo/core/tasks.py

```diff
@@ -12,38 +12,60 @@
 
 from wagtail.contrib.settings.context_processors import SettingsProxy
 from wagtail.wagtailcore.models import Site
-
+from django.utils import timezone
 
 IMPORT_EMAIL_TEMPLATE = "core/content_import/import_email.html"
 VALIDATE_EMAIL_TEMPLATE = "core/content_import/validate_email.html"
 
 
 @task(ignore_result=True)
 def rotate_content():
+    """ this method gets the parameters that are needed for rotate_latest
+    and rotate_featured_in_homepage methods, and calls them both"""
+    # getting the content rotation settings from site settings
     main_lang = SiteLanguage.objects.filter(is_main_language=True).first()
     main = Main.objects.all().first()
     index = SectionIndexPage.objects.live().first()
     site = Site.objects.get(is_default_site=True)
     settings = SettingsProxy(site)
     site_settings = settings['core']['SiteSettings']
+    day = datetime.today().weekday()
+    # creates a days of the week list
+    days = [
+        site_settings.monday, site_settings.tuesday, site_settings.wednesday,
+        site_settings.thursday, site_settings.friday, site_settings.saturday,
+        site_settings.sunday]
+    # calls the two rotate methods with the necessary params
     if main and index:
-        if site_settings.content_rotation_time == datetime.now().hour:
-            if site_settings.content_rotation:
-                rotate_latest(main_lang, index, main)
-            rotate_featured_in_homepage(main_lang)
-
-
-def rotate_latest(main_lang, index, main):
-    random_article = ArticlePage.objects.live().filter(
-        featured_in_latest=False, languages__language__id=main_lang.id
-    ).descendant_of(index).order_by('?').first()
-    if random_article:
-        random_article.featured_in_latest = True
-        random_article.save_revision().publish()
-
-        article = main.latest_articles().last()
-        article.featured_in_latest = False
-        article.save_revision().publish()
+        rotate_latest(main_lang, index, main, site_settings, days, day)
+        rotate_featured_in_homepage(main_lang)
+
+
+def rotate_latest(main_lang, index, main, site_settings, days, day):
+    """This rotates all the articles that have been marked as
+    featured_in_latest. It checks whether current date falls within the set
+    date range for content rotation. It then checks whether the current weekday
+    is set to rotate, and then rotates an articles for each hour the admin has
+    set."""
+    if site_settings.content_rotation_start_date < timezone.now() \
+            < site_settings.content_rotation_end_date:
+        # checks if the current weekday is set to rotate
+        if days[day - 1]:
+            for time in site_settings.time:
+                if str(time)[:2] == str(datetime.now().hour):
```

Comment:
depending on the value of `time`, you should be able to get to the `hour` without using `str`